src/unit/test_listpack.c

```diff
@@ -0,0 +1,976 @@
+    lp = lpAppend(lp, (unsigned char*)intlist[4], strlen(intlist[4]));
+    lp = lpAppend(lp, (unsigned char*)intlist[5], strlen(intlist[5]));
+    return lp;
+}
+
+static long long usec(void) {
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    return (((long long)tv.tv_sec)*1000000)+tv.tv_usec;
+}
+
+static void stress(int pos, int num, int maxsize, int dnum) {
+    int i, j, k;
+    unsigned char *lp;
+    char posstr[2][5] = { "HEAD", "TAIL" };
+    long long start;
+    for (i = 0; i < maxsize; i+=dnum) {
+        lp = lpNew(0);
+        for (j = 0; j < i; j++) {
+            lp = lpAppend(lp, (unsigned char*)"quux", 4);
+        }
+
+        /* Do num times a push+pop from pos */
+        start = usec();
+        for (k = 0; k < num; k++) {
+            if (pos == 0) {
+                lp = lpPrepend(lp, (unsigned char*)"quux", 4);
+            } else {
+                lp = lpAppend(lp, (unsigned char*)"quux", 4);
+
+            }
+            lp = lpDelete(lp, lpFirst(lp), NULL);
+        }
+        printf("List size: %8d, bytes: %8zu, %dx push+pop (%s): %6lld usec\n",
+               i, lpBytes(lp), num, posstr[pos], usec()-start);
+        lpFree(lp);
+    }
+}
+
+static unsigned char *pop(unsigned char *lp, int where) {
+    unsigned char *p, *vstr;
+    int64_t vlen;
+
+    p = lpSeek(lp, where == 0 ? 0 : -1);
+    vstr = lpGet(p, &vlen, NULL);
+    if (where == 0)
+        printf("Pop head: ");
+    else
+        printf("Pop tail: ");
+
+    if (vstr) {
+        if (vlen && fwrite(vstr, vlen, 1, stdout) == 0) perror("fwrite");
+    } else {
+        printf("%lld", (long long)vlen);
+    }
+
+    printf("\n");
+    return lpDelete(lp, p, &p);
+}
+
+static int randstring(char *target, unsigned int min, unsigned int max) {
+    int p = 0;
+    int len = min+rand()%(max-min+1);
+    int minval, maxval;
+    switch(rand() % 3) {
+    case 0:
+        minval = 0;
+        maxval = 255;
+    break;
+    case 1:
+        minval = 48;
+        maxval = 122;
+    break;
+    case 2:
+        minval = 48;
+        maxval = 52;
+    break;
+    default:
+        TEST_ASSERT(NULL);
+    }
+
+    while(p < len)
+        target[p++] = minval+rand()%(maxval-minval+1);
+    return len;
+}
+
+static int verifyEntry(unsigned char *p, unsigned char *s, size_t slen) {
+    TEST_ASSERT(lpCompare(p, s, slen));
+    return 0;
+}
+
+static int lpValidation(unsigned char *p, unsigned int head_count, void *userdata) {
+    UNUSED(p);
+    UNUSED(head_count);
+
+    int ret;
+    long *count = userdata;
+    ret = lpCompare(p, (unsigned char *)mixlist[*count], strlen(mixlist[*count]));
+    (*count)++;
+    return ret;
+}
+
+int test_listpack(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+
+    int i;
+    unsigned char *lp, *p, *vstr;
+    int64_t vlen;
+    unsigned char intbuf[LP_INTBUF_SIZE];
+    int accurate = (flags & UNIT_TEST_ACCURATE);
+
+    TEST("Create int list") {
```

Comment:
I think It would be better if we break these `TEST` into different test case functions. For example, something like:

```c
int test_listpackCreateIntList(int argc, char **argv, int flags)
```

Similarly for other places as well, when possible.